src/CloudFlare/IpRewrite.php

```diff
@@ -69,38 +82,32 @@ public static function reset()
     protected static function rewrite()
     {
         // only should be run once per page load
-        if (self::$is_cf !== NULL) {
+        if (self::$is_loaded) {
             return;
         }
-        
-        // Set this based on the presence of the header, so it is true even if IP has already been rewritten
-        self::$is_cf = isset($_SERVER["HTTP_CF_CONNECTING_IP"]) ? TRUE : FALSE;
-        
+        self::$is_loaded = true;
+
+        $is_cf = self::isCloudFlare();
+        if (!$is_cf) {
+            return;
+        }
+
         // Store original remote address in $original_ip
-        self::$original_ip = isset($_SERVER["REMOTE_ADDR"]) ? $_SERVER["REMOTE_ADDR"] : FALSE;
+        if (!isset($_SERVER['REMOTE_ADDR'])) {
+            return;
+        }
+        self::$original_ip = $_SERVER['REMOTE_ADDR'];
 
```

Comment:
I didn't want to put any side effects inside `getOriginalIp()` so I chose to explicitly set the `original_ip` to `$_SERVER['REMOTE_ADDR']` inside `rewrite` since it's `rewrite`'s job to do this.